reana_job_controller/kubernetes_job_manager.py

```diff
@@ -32,6 +33,8 @@
 class KubernetesJobManager(JobManager):
     """Kubernetes job management."""
 
+    MAX_JOB_RESTARTS = 3
```

Comment:
We already have `MAX_JOB_RESTARTS` in the global `config.py`. (1) One variable may be enough and use it for all backends. (2) Perhaps we should name it like `MAX_NUM_RESUBMISSIONS` to make it clear that this is just for job submission, i.e. retrying the submit if the job submission failed for a reason or another, and not retrying the run when the user job was submitted fine but failed for a reason or another (where we should exit immediately without retrying).